src/authRemover.ts

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { AsyncOptionalCreatable } from '@salesforce/kit';
+import { Nullable } from '@salesforce/ts-types';
+import * as path from 'path';
+import { AuthInfo } from './authInfo';
+import { Aliases } from './config/aliases';
+import { AuthInfoConfig } from './config/authInfoConfig';
+import { Config } from './config/config';
+import { ConfigAggregator } from './config/configAggregator';
+import { Logger } from './logger';
+import { Messages } from './messages';
+import { SfdxError } from './sfdxError';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/core', 'auth');
+
+export type AuthConfigs = Map<string, AuthInfoConfig>;
+
+/**
+ * Handles the removing of authorizations, which includes deleting the auth file
+ * in the global .sfdx folder, deleting any configs that are associated with the username/alias,
+ * and deleting any aliases associated with the username
+ *
+ * ```
+ * const remover = await AuthRemover.create();
+ * await remover.removeAuth('example@mycompany.com');
+ * ```
+ *
+ * ```
+ * const remover = await AuthRemover.create();
+ * const authConfigs = await remover.findAuthConfigs(
+ *  example@mycompany.com
+ * );
+ * const usernames = [...authConfigs.keys()];
+ * for (const username of usernames) {
+ *   await remover.removeAuth(username);
+ * }
+ * ```
+ */
+export class AuthRemover extends AsyncOptionalCreatable {
```

Comment:
Nice job with this class. I like the API.

Reply:
Yeah, this is awesome!